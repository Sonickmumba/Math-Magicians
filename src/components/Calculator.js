import React, { useState } from 'react';
import '../App.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [object, setObject] = useState({
    next: null,
    total: '0',
    operation: null,
  });

  const handleClick = (e) => {
    e.preventDefault();
    const { next, total, operation } = calculate(object, e.target.name);

    if (next === null && total === null) {
      setObject({ next, total: '0', operation });
    } else {
      setObject({ next, total, operation });
    }
  };

  const { total, next } = object;
  const btnNames = [
    'AC',
    '+/-',
    '%',
    'รท',
    '7',
    '8',
    '9',
    'x',
    '4',
    '5',
    '6',
    '-',
    '1',
    '2',
    '3',
    '+',
    '0',
    '.',
    '=',
  ];

  return (
    <div className="calculator-main-container">
      <div className="calculator-title-container">
        <h2 className="calculator-title">Let us do some math!</h2>
      </div>
      <div className="calculator-grid-container">
        { next ? (
          <div className="output-result">
            { next }
          </div>
        ) : (
          <div className="output-result">
            { total }
          </div>
        )}
        { btnNames.map((btnName) => (
          <button
            onClick={handleClick}
            name={btnName}
            type="button"
            className={`btn ${btnName === '0' ? 'zero-button' : ''}
            ${btnName === '=' ? 'button-operation' : ''}
            ${btnName === 'รท' ? 'button-operation' : ''}
            ${btnName === 'x' ? 'button-operation' : ''}
            ${btnName === '-' ? 'button-operation' : ''}
            ${btnName === '+' ? 'button-operation' : ''}`}
            key={btnName}
          >
            { btnName }
          </button>
        ))}
      </div>
    </div>
  );
};

export default Calculator;
