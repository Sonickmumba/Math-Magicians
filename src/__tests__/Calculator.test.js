import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('Test for Calculator Component', () => {
  render(<Calculator />);
  const linkElement = screen.getByText(/Let us do some math/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Test for Calculator component', () => {
  it('Calculator Component renders correctly', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
