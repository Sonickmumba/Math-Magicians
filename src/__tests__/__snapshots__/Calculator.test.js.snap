// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Calculator component Calculator Component renders correctly 1`] = `
<div
  className="calculator-main-container"
>
  <div
    className="calculator-title-container"
  >
    <h2
      className="calculator-title"
    >
      Let us do some math!
    </h2>
  </div>
  <div
    className="calculator-grid-container"
  >
    <div
      className="output-result"
    >
      0
    </div>
    <button
      className="btn 
            
            
            
            
            "
      name="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn 
            
            button-operation
            
            
            "
      name="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn 
            
            
            button-operation
            
            "
      name="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn 
            
            
            
            button-operation
            "
      name="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn 
            
            
            
            
            button-operation"
      name="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn zero-button
            
            
            
            
            "
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn 
            button-operation
            
            
            
            "
      name="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Unit Test For Calculator component Calculator Component renders correctly 1`] = `
<div
  className="calculator-main-container"
>
  <div
    className="calculator-title-container"
  >
    <h2
      className="calculator-title"
    >
      Let us do some math!
    </h2>
  </div>
  <div
    className="calculator-grid-container"
  >
    <div
      className="output-result"
    >
      0
    </div>
    <button
      className="btn 
            
            
            
            
            "
      name="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn 
            
            button-operation
            
            
            "
      name="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn 
            
            
            button-operation
            
            "
      name="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn 
            
            
            
            button-operation
            "
      name="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn 
            
            
            
            
            button-operation"
      name="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn zero-button
            
            
            
            
            "
      name="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn 
            
            
            
            
            "
      name="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn 
            button-operation
            
            
            
            "
      name="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
